### PR TITLE
Dev spi carter

### PR DESCRIPTION
**Pull request by cshaul**
Additional fifo functionality was added. Code cleanup performed. Ready for algorithm development.

### CODE REVIEW

File: project/LIS2DH12.c [hunk 1]
```diff
@@ -63,14 +63,15 @@ typedef struct {
...
+lis2dh12_instance_t accel_inst = {0}; // Instance of the LIS2DH12
 
 #define SPI_INSTANCE  0 //SPI instance index
 const nrf_drv_spi_t spi = NRF_DRV_SPI_INSTANCE(SPI_INSTANCE); //SPI instance
 volatile bool spi_xfer_done; //Flag used to indicate that SPI instance completed the transfer
 
 #define SPI_BUFFER_LENGTH (192)
 uint8_t m_tx_buf[SPI_BUFFER_LENGTH]; //Tx buffer
-uint8_t m_rx_buf[SPI_BUFFER_LENGTH]; //Rx buffer
+uint8_t m_rx_buf[SPI_BUFFER_LENGTH + 1]; //Rx buffer
```
Comment (funkna) [inline project/LIS2DH12.c:74]: Why is this necessary?

File: project/LIS2DH12.c [hunk 2]
```diff
@@ -64,15 +64,11 @@ typedef struct {
...
-/*NOTE: Uncomment these lines for more debug info*/
-//#define SPI_DEBUG_INFO
-//#define ACCEL_DEBUG_INFO
-
 #define SPI_INSTANCE  0 //SPI instance index
 const nrf_drv_spi_t spi = NRF_DRV_SPI_INSTANCE(SPI_INSTANCE); //SPI instance
 volatile bool spi_xfer_done; //Flag used to indicate that SPI instance completed the transfer
 
-#define SPI_BUFFER_LENGTH (128)
+#define SPI_BUFFER_LENGTH (192)
```
Comment (funkna) [inline project/LIS2DH12.c:72]: How'd you get this number?  If it's an EXACT number, maybe add 10% or round to the nearest 2^x so we have room for error and unexpected situations.

File: project/LIS2DH12.c (hunk 1 above)
Comment (AaronFarq) [reply project/LIS2DH12.c:74]: This is necessary because if we are receiving 192 bytes, then the rx buffer needs to be at least 193 bytes, because received data starts at index 1 and not index 0. If we added 10% or if we just made it 256 instead, this wouldn't be necessary, but then we would have to adjust the read function. I am good either way
